### DISCUSSION TITLE
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'csrf')

### DISCUSSION THREAD

**Discussion by rohait0717** [livewire/livewire#6603] [1. Help] [answered]
While navigating to a page using the 'wire:navigate' attribute on the live server, I encounter an error. This error arises when the page contains a Livewire lazy component that requires requesting data from the database. However, if I refresh the page, the issue is resolved and the component functions properly

"Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'csrf')
    at getCsrfToken (livewire.js?id=11c49d7e:66:31)
    at livewire.js?id=11c49d7e:83:227
    at queueNewRequestAttemptsWhile (livewire.js?id=11c49d7e:94:81)
    at sendRequestToServer (livewire.js?id=11c49d7e:83:68)
    at livewire.js?id=11c49d7e:80:214
    at livewire.js?id=11c49d7e:82:45"

**Comment by rohait0717** [comment, marked as answer]
Cloudflare Rocket Loader is responsible for causing this error.

**Comment by bretterer** [comment]
What was your solution here?  I don't have cloudflare as i'm in local development with this

**Reply by FinnPaes** [reply]
In my case this happened because I first had a ``<div>`` and afterwards I changed that to a form tag. Didn't close it with ``</form>`` but forgot and kept ``</div>`` :

```html
<form wire:submit="submit">
<!-- form fields here -->
</div>
```

This will also throw that error!
